src/promptflow/tests/executor/unittests/_utils/test_dataclass_serializer.py

```diff
@@ -0,0 +1,204 @@
+import pytest
+from datetime import datetime
+from dataclasses import dataclass
+from typing import Dict, List
+from promptflow._core.generator_proxy import GeneratorProxy
+from promptflow._utils.dataclass_serializer import \
+    get_type, serialize, deserialize_dataclass, deserialize_value, assertEqual
+from promptflow.contracts.run_info import RunInfo, Status
+from promptflow._core.connection_manager import ConnectionManager
+from promptflow.storage.run_records import NodeRunRecord
+from unittest.mock import patch, Mock
+import sys
+
+
+def get_connection_dict():
+    return {
+        "azure_open_ai_connection": {
+            "type": "AzureOpenAIConnection",
+            "value": {
+                "api_key": "<azure-openai-key>",
+                "api_base": "https://gpt-test-eus.openai.azure.com/",
```

Comment:
It is not a good idea to use a real api_base..

Reply:
Replaced api_base with "<aoai-api-endpoint>".